test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -348,25 +364,27 @@ specification =
     unsafeDecode = either ( error . (msg <>) . show) id . Yaml.decodeEither'
     msg = "Whoops! Failed to parse or find the api specification document: "
 
--- | Ad-hoc 'ToSchema' instance for the 'Wallet' definition: we simply look it
--- up from the specification.
 instance ToSchema Wallet where
-    declareNamedSchema _ = case specification ^. definitions . at "Wallet" of
-        Nothing ->
-            error "unable to find the definition for 'Wallet' in the spec"
-        Just schema ->
-            return $ NamedSchema (Just "Wallet") schema
+    declareNamedSchema _ = declareSchemaForDefinition "Wallet"
 
--- | Ad-hoc 'ToSchema' instance for the 'WalletPostData' definition: we simply
--- look it up from the specification.
 instance ToSchema WalletPostData where
-    declareNamedSchema _ =
-        case specification ^. definitions . at "WalletPostData" of
-            Nothing -> error
-                "unable to find the definition for 'WalletPostData' in \
-                \the spec"
-            Just schema ->
-                return $ NamedSchema (Just "WalletPostData") schema
+    declareNamedSchema _ = declareSchemaForDefinition "WalletPostData"
+
+instance ToSchema WalletPutData where
+    declareNamedSchema _ = declareSchemaForDefinition "WalletPutData"
+
+instance ToSchema WalletPutPassphraseData where
+    declareNamedSchema _ = declareSchemaForDefinition "WalletPutPassphraseData"
+
+-- | Utility function to provide an ad-hoc 'ToSchema' instance for a definition:
+-- we simply look it up within the Swagger specification.
+declareSchemaForDefinition :: T.Text -> Declare (Definitions Schema) NamedSchema
+declareSchemaForDefinition name =
+    case specification ^. definitions . at name of
+        Nothing -> error $
+            "unable to find the definition for " <> show name <> " in the spec"
+        Just schema ->
+            return $ NamedSchema (Just name) schema
```

Comment:
I think I'd even push it further and make it all "automagic" via a `Typeable` instance :thinking:  ?